services/storage/src/simcore_service_storage/s3.py

```diff
@@ -25,24 +24,22 @@ async def setup_s3_client(app):
     storage_s3_settings = storage_settings.STORAGE_S3
     assert storage_s3_settings  # nosec
 
-    async with AsyncExitStack() as exit_stack:
-        client = None
-        async for attempt in AsyncRetrying(
-            wait=wait_fixed(RETRY_WAIT_SECS),
-            before_sleep=before_sleep_log(log, logging.WARNING),
-            reraise=True,
-        ):
-            with attempt:
-                client = await StorageS3Client.create(
-                    exit_stack,
-                    storage_s3_settings,
-                    storage_settings.STORAGE_S3_CLIENT_MAX_TRANSFER_CONCURRENCY,
-                )
-                log.info(
-                    "S3 client %s successfully created [%s]",
-                    f"{client=}",
-                    json.dumps(attempt.retry_state.retry_object.statistics),
-                )
+    client = None
+    async for attempt in AsyncRetrying(
+        wait=wait_fixed(RETRY_WAIT_SECS),
+        before_sleep=before_sleep_log(log, logging.WARNING),
+        reraise=True,
+    ):
+        with attempt:
+            client = await StorageS3Client.create(
+                storage_s3_settings,
+                storage_settings.STORAGE_S3_CLIENT_MAX_TRANSFER_CONCURRENCY,
+            )
+            log.info(
+                "S3 client %s successfully created [%s]",
+                f"{client=}",
+                json.dumps(attempt.retry_state.retry_object.statistics),
+            )
```

Comment:
is client.close not missing here after the yield??

Reply:
yep. careful! Also if it raises, be sure that no client is created (e.g. can it raise after the client is created but e.g. cannot connect?)